### PR TITLE
gcp secure instances

### PR DESCRIPTION
**Pull request by AlexBMes**
Using specs for confidential vm creates: persistent disk, instance template and managed instance group for listed zones

### CODE REVIEW

**Review by gbmeuk — CHANGES_REQUESTED** [review]
I'm not sure how much value we're adding by having a module for the disks I realise that we will be able to apply the guest os features, it's just that, as you  proved, we don't currently need to use it and it will probably be natively available soon.

I also think that we can go with a single instance template and instance where we have a 'security level' value which configures them to an agreed standard. Then have a secure swarm composite module which utilises it. 

Let's discuss on Monday

File: gcp/secure_swarm_node/locals.tf [hunk 1]
```diff
@@ -1,4 +1,12 @@
 locals {
   service_account_email = var.service_account_email == "" ? "${data.google_project.default.number}@-compute@developer.gserviceaccount.com" : var.service_account_email
+
+  blue_instance_template = lookup(var.blue_instance_template, "security_level", null) == null ? merge(
+    var.blue_instance_template, { security_level = var.security_level
+  }) : contains(["secure-1", "confidential-1"], var.blue_instance_template["security_level"]) ? var.blue_instance_template : merge(var.blue_instance_template, { security_level = "confidential-1" })
+
+  green_instance_template = lookup(var.blue_instance_template, "security_level", null) == null ? merge(
```
[the remark this answers is not in the document]
Comment (AlexBMes) [reply gcp/secure_swarm_node/locals.tf]: Yes they were the same accidentally, have updated the green to use `var.green_instance_template`. Had also forgotten to implement these variables so have updated the `secure_instance_template_blue/green` modules to use the local variables

File: gcp/secure_swarm_node/main.tf [hunk 2]
```diff
@@ -1,3 +1,7 @@
+terraform {
+  experiments = [module_variable_optional_attrs]
```
[the remark this answers is not in the document]
Comment (AlexBMes) [reply gcp/secure_swarm_node/main.tf:3]: Updated the documentation and minimum version to v0.14.0

**Review by gbmeuk — APPROVED** [review]
Approved but @danperezsan please review as well so as to use for the root module creation for client

**Review by danperezsan — APPROVED** [review]

**Pull request merged**